### PR TITLE
Lu 7188 - added dead letter functionality

### PR DESCRIPTION
**Pull request by jamiestephenkerr**

### CODE REVIEW

File: BlaiseCaseMonitor.Tests/Services/QueueServiceTests.cs [hunk 1]
```diff
@@ -1,42 +1,85 @@
...
+        private Mock<ILog> _loggerMock;
         private Mock<IConfigurationProvider> _configurationProviderMock;
+        private Mock<IMessageHandler> _messageHandlerMock;
         private Mock<IFluentQueueApi> _queueProviderMock;
 
         private readonly string _projectId;
-        private readonly string _topicId;
+        private readonly string _publishTopicId;
+        private readonly string _subscriptionTopicId;
+        private readonly string _subscriptionId;
```
Comment (anax32) [inline BlaiseCaseMonitor.Tests/Services/QueueServiceTests.cs:20]: is there a better naming for these two variables? Unless I'm out of the loop, just feels a bit..

Comment (jamiestephenkerr) [reply BlaiseCaseMonitor.Tests/Services/QueueServiceTests.cs:20]: It is a stop gap :)

File: BlaiseCaseMonitor/Services/InitialiseService.cs [hunk 2]
```diff
@@ -9,45 +9,44 @@ public class InitialiseService : IInitialiseService
...
+            }
+
+            _logger.Info($"Nisra case monitor service started on '{_configurationProvider.VmName}'");
         }
 
         public void Stop()
         {
-            _logger.Info("Blaise Case Monitor service stopped.");
-        }
+            _logger.Info($"Stopping Nisra Case processing service on '{_configurationProvider.VmName}'");
```
Comment (anax32) [inline BlaiseCaseMonitor/Services/InitialiseService.cs:45]: Nisra case processing?

File: BlaiseCaseMonitor/Services/InitialiseService.cs [hunk 3]
```diff
@@ -9,45 +9,44 @@ public class InitialiseService : IInitialiseService
...
+            try
+            {
+                _queueService.Subscribe(_messageHandler);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex);
+            }
+
+            _logger.Info($"Nisra case monitor service started on '{_configurationProvider.VmName}'");
```
Comment (anax32) [inline BlaiseCaseMonitor/Services/InitialiseService.cs:40]: nisra case monitor here, but processing below.

**Pull request merged**